src/sinks/new_relic_logs.rs

```diff
@@ -13,12 +13,20 @@ use indexmap::IndexMap;
 use serde::{Deserialize, Serialize};
 use snafu::Snafu;
 
+// New Relic Logs API accepts payloads up to 1MB (10^6 bytes)
+const MAX_PAYLOAD_SIZE: usize = 1000000 as usize;
+
 #[derive(Debug, Snafu)]
 enum BuildError {
     #[snafu(display(
         "Missing authentication key, must provide either 'license_key' or 'insert_key'"
     ))]
     MissingAuthParam,
+    #[snafu(display(
+        "Too high batch max size. The value must be {} bytes or less",
+        max_size
```

Comment:
Nit: I think you could leave out the parameter here and just use the defined constant, `MAX_PAYLOAD_SIZE`, as the parameter to the format string.

You could do maybe do something like:

```rust
     #[snafu(display(
        "Batch size of {} too high. The value must be {} bytes or less",
        batch_size,
        MAX_PAYLOAD_SIZE
     ))]
    BatchMaxSize { batch_size: usize },
```

Reply:
Yes, considering this error is used only for this sink, the value is essentially a constant, rather than a parameter.